Task5/api/controllers/usersController.js

```diff
@@ -0,0 +1,125 @@
+// usersController.js
+const { v4: uuidv4 } = require('uuid');
+
+const users = [];
+const hobbies = []
+
+module.exports = {
+    getAllUsers: (req, res) => {
+        // Set headers
+        res.writeHead(200, {
+            'Content-Type': 'application/json',
+            'Cache-Control': 'public, max-age=3600'
+        });
+
+        // Send the response body
+        res.end(JSON.stringify({ data: users, error: null }));
+    },
+    createUser: (req, res, data) => {
+        const ID = uuidv4();
+
+        const newUser = {
+            user: {
+                id: ID,
+                name: data.name,
+                email: data.email
+            },
+            links: {
+                self: `/api/users/${ID}`,
+                hobbies: `/api/users/${ID}/hobbies`
+            }
+        };
+
+        const newHobbies = {
+            userId: ID,
+            hobbies: [],
+            links: {
+                self: `/api/users/${ID}/hobbies`,
+                user: `/api/users/${ID}`
+            }
+        }
+
+        users.push(newUser);
+        hobbies.push(newHobbies);
+
+        res.statusCode = 200;
+        res.setHeader("Content-Type", "application/json");
+        res.end(JSON.stringify({ data: newUser, error: null }));
```

Comment:
make function that does this so you dont need to write this 3 lines in every method